File: module-git/src/main/java/com/timecat/module/git/sgit/repo/tasks/repo/RebaseTask.java
package com.timecat.module.git.sgit.repo.tasks.repo;

import com.timecat.module.git.R;
import com.timecat.module.git.sgit.database.models.Repo;
import com.timecat.module.git.sgit.exception.StopTaskException;

public class RebaseTask extends RepoOpTask {

    public String mUpstream;
    private AsyncTaskPostCallback mCallback;

    public RebaseTask(Repo repo, String upstream, AsyncTaskPostCallback callback) {
        super(repo);
        mUpstream = upstream;
        mCallback = callback;
        setSuccessMsg(R.string.git_success_rebase);
    }

    @Override
    protected Boolean doInBackground(Void... params) {
        return rebase();
    }

    protected void onPostExecute(Boolean isSuccess) {
        super.onPostExecute(isSuccess);
        if (mCallback != null) {
            mCallback.onPostExecute(isSuccess);
        }
    }

    public boolean rebase() {
        try {
            mRepo.getGit().rebase().setUpstream(mUpstream).call();
        } catch (StopTaskException e) {
            return false;
        } catch (Throwable e) {
            setException(e);
            return false;
        }
        return true;
    }
}

File: module-git/src/main/java/com/timecat/module/git/tasks/RebaseTask.kt
package com.timecat.module.git.tasks

import com.timecat.module.git.R
import com.timecat.module.git.sgit.database.models.Repo
import com.timecat.module.git.utils.StopTaskException

class RebaseTask(repo: Repo, var mUpstream: String, private val mCallback: AsyncTaskPostCallback?) : RepoOpTask(repo) {
    init {
        setSuccessMsg(R.string.git_success_rebase)
    }

    override fun doInBackground(vararg params: Void): Boolean {
        return rebase()
    }

    override fun onPostExecute(isSuccess: Boolean) {
        super.onPostExecute(isSuccess)
        mCallback?.onPostExecute(isSuccess)
    }

    fun rebase(): Boolean {
        try {
            val git = mRepo.getGit() ?: return false
            git.rebase().setUpstream(mUpstream).call()
        } catch (e: StopTaskException) {
            return false
        } catch (e: Throwable) {
            setException(e)
            return false
        }
        return true
    }
}